File: app/src/main/java/c/m/storyapp/authentication_check/domain/repository/AuthenticationCheckRepository.kt
package c.m.storyapp.authentication_check.domain.repository

import c.m.storyapp.authentication_check.domain.model.AuthenticationStatus
import c.m.storyapp.authentication_check.domain.model.AuthenticationToken
import c.m.storyapp.common.util.Resource
import c.m.storyapp.common.util.SimpleResource
import kotlinx.coroutines.flow.Flow

interface AuthenticationCheckRepository {
    suspend fun saveToken(token: String): Flow<SimpleResource>
    suspend fun checkAuthentication(): Flow<Resource<AuthenticationStatus>>
    suspend fun getToken(): Flow<Resource<AuthenticationToken>>
}

File: lib/authentication_check/domain/repository/authentication_check_repository.dart
import 'package:story_app/authentication_check/domain/model/authentication_status.dart';
import 'package:story_app/authentication_check/domain/model/authentication_token.dart';
import 'package:story_app/common/util/resource.dart';

abstract class AuthenticationCheckRepository {
  Future<void> saveToken(String token);
  Future<Resource<AuthenticationStatus>> checkAuthentication();
  Future<Resource<AuthenticationToken>> getToken();
}
